test/test.sh

```diff
@@ -1272,6 +1272,30 @@ test_no_database_auth_user() {
 	return 0
 }
 
+test_no_database_scram_auth_success() {
+	# Testing what happens on successful SCRAM auth connection to non-existent DB
+	# Segfaults have been seen after mock authentication was put in place
+
+	admin "set auth_type='md5'"
+	PGPASSWORD=foo psql -X -U scramuser1 -d nosuchdb1 -c "select 1" && return 1
+	grep -F "no such database: nosuchdb1" $BOUNCER_LOG || return 1
+
+    return 0
+}
+
+test_no_database_md5_auth_success() {
+	# Testing what happens on successful SCRAM auth connection to non-existent DB
+	# Segfaults have been seen after mock authentication was put in place
+
+	admin "set auth_type='md5'"
```

Comment:
This test didn't fail before.  I agree that it makes sense for completeness, but maybe swap the auth_type here too (so use `scram-256` instead of md5). Or add all 4 different combinations:
1. scramuser1 + md5
2. scramuser1 + scram-256
3. muser1 + md5
4. muser1 + scram-256

Reply:
Good suggestion. Added all combinations here except for `muser1 + scram` since that failed authentication (as its supposed to).